### DISCUSSION TITLE
Confidence intervals

### DISCUSSION THREAD

**Discussion by RionIwasaki** [WeirongChen/DeepEdge#3] [Q&A] [answered]
Have a few questions on confidence intervals, which show up when you select "NeuralNet" or "NN + Snake" option.
(1) Even with the presence of confidence intervals, are the ultimate X and Y coordinates for each contour still based on red points (or light blue ones if you put anchor points manually)?
(2) It seems to me that with "NeuralNet", the anchor points are placed in the middle of the confidence interval (in the middle of the top and bottom edges of the purple band). Is this observation correct? In contrast, I suppose "NN + Snake" detects the edges and places anchor points based on the confidence intervals.

**Comment by WeirongChen** [comment, marked as answer]
Thanks for those questions!
The inference of contour (under the hood) is in the following steps:
1. For every pixel in an image, the model predicts the probability of having an edge contour on this pixel. 
2. A cut-off threshold (NN Prob. Thresh. default=0.4) is selected to create a region of interest that encompasses tongue contours (the purple line) (i.e., confidence region/interval in your question.)
3. A mid-line through the confidence region is drawn as NN-predicted contour. (Thus, your hunch in the second question is correct!)
4. (If NN+SNAKE is selected) SNAKE algorithm snaps the NN-predicted contour to the lower edge of the visible contour on the image.

For your first question, the x-y coordinates in the output file are exactly what you see in the anchor points (Blue: model-generated; Red: user inputs) on the screen.
